pydantic/_internal/_generate_schema.py

```diff
@@ -1480,6 +1462,37 @@ def json_schema_update_func(
             return maybe_updated_schema
         return original_schema
 
+    def apply_single_annotation_json_schema(
```

Comment:
I had to break this out as a separate function in order to ensure the callable JSON schema ran _after_ all modifications to the core schema that influence the json schema are performed. Otherwise, for example, you can remove the `default`.